HyperVExtension/src/HyperVExtension/Helpers/DevSetupAgentDeploymentHelper.cs

```diff
@@ -0,0 +1,306 @@
+            $isDirectory = $true
+        }
+        elseif (Test-Path -Path $Path -PathType 'Leaf')
+        {
+            if ($Path -match '\.(cab)$')
+            {
+                $isCab = $true
+            }
+            elseif (-not $Path -match '\.(zip)$')
+            {
+                throw ""Only .cab and .zip files are supported""
+            }
+            $inputFileName = Split-Path -Path $Path -Leaf
+        }
+        else
+        {
+            throw ""$Path does not exist""
+        }
+
+
+        $DevSetupAgentConst = ""DevSetupAgent""
+        $DevSetupEngineConst = ""DevSetupEngine""
+        $session = $Session
+
+        $guestTempDirectory = Invoke-Command -Session $session -ScriptBlock { $env:temp }
+        
+        [string] $guid = [System.Guid]::NewGuid()
+        $guestUnpackDirectory = Join-Path -Path $guestTempDirectory -ChildPath $guid
+        $guestDevSetupAgentTempDirectory = Join-Path -Path $guestUnpackDirectory -ChildPath $DevSetupAgentConst
+
+        Write-Host ""Creating VM temporary folder $guestUnpackDirectory""
+        Write-Progress -Activity $activity -Status ""Creating VM temporary folder $guestUnpackDirectory"" -PercentComplete 10
+        Invoke-Command -Session $session -ScriptBlock { New-Item -Path ""$using:guestUnpackDirectory"" -ItemType ""directory"" }
+
+        if ($isDirectory)
+        {
+            $destinationPath = $guestDevSetupAgentTempDirectory
+        }
+        else
+        {
+            $destinationPath = $guestUnpackDirectory
+        }
+
+        Write-Host ""Copying $Path to VM $destinationPath""
+        Write-Progress -Activity $activity -Status ""Copying DevSetupAgent to VM $destinationPath"" -PercentComplete 15
+        Copy-Item -ToSession $session -Recurse -Path $Path -Destination $destinationPath
+
+
+        Invoke-Command -Session $session -ScriptBlock {
+            $ErrorActionPreference = ""Stop""
+
+            try
+            {
+                $guestDevSetupAgentPath = Join-Path -Path $Env:Programfiles -ChildPath $using:DevSetupAgentConst
+
+                # Stop and remove previous version of DevSetupAgent service if it exists
+                $service = Get-Service -Name $using:DevSetupAgentConst -ErrorAction SilentlyContinue
+                if ($service -ne $null)
+                {
+                    $serviceWMI = Get-WmiObject -Class Win32_Service -Filter ""Name='$using:DevSetupAgentConst'""
+                    $existingServicePath = $serviceWMI.Properties[""PathName""].Value
+                    if ($existingServicePath)
+                    {
+                        $guestDevSetupAgentPath = Split-Path $existingServicePath -Parent
+                    }
+                    
+                    try
+                    {
+                        Write-Host ""Stopping DevSetupAgent service""
+                        Write-Progress -Activity $using:activity -Status ""Stopping DevSetupAgent service $destinationPath"" -PercentComplete 30

```

Comment:
I guess in this case we're hoping the service isn't in a paused/pausing state?

Reply:
You can't pause it. It doesn't have "paused" functionality.